src/lib.rs

```diff
@@ -405,25 +417,24 @@ fn alloc_size<T>(cap: usize) -> usize {
         .unwrap_cap_overflow();
 
     // Ok now we can turn it back into a usize (don't need to worry about negatives)
-    final_size as usize
+    final_size
+        .try_into()
+        .expect("cannot allocate object larger than isize")
```

Comment:
Use of `try_into` for these casts seems like it could cause a perf regression? Which seems a bit unnecessary given that we can be confident that they will always succeed.

Do you have a motivation for changing them other than that the lint says to?